olive/model/__init__.py

```diff
@@ -588,7 +588,7 @@ def prepare_session(
         execution_providers: Union[str, List[str]] = None,
         rank: Optional[int] = None,
     ):
-        return self.load_model().eval()
+        return self.load_model(device).eval()
```

Comment:
I think the model.load_model signature doesn't need to be changed after the model is loaded just by changing as the following line.
```python
model = self.load_model()
model.to("cuda") if device == Device.GPU else "cpu"
return model.eval()

```

Reply:
with above change, the hf_utils.py doesn't need to be changed since currently it only handle hf models.

Reply:
Another consideration is to make the load_model signature be consistent with other classes.

Reply:
I and Jambay had a similar discussion above. We decide to revert this change since evaluator applies device and it works for both hf model and pytorch model.